helper-scripts/backup_and_restore.sh

```diff
@@ -1,331 +1,621 @@
-          ${SQLIMAGE} /bin/sh -c "mysqld --skip-grant-tables & \
-          until mysqladmin ping; do sleep 3; done && \
-          echo Restoring... && \
-          gunzip < backup/backup_mysql.gz | mysql -uroot && \
-          mysql -uroot -e SHUTDOWN;"
-        elif [[ -f "${RESTORE_LOCATION}/backup_mariadb.tar.gz" ]]; then
-        docker run --name mailcow-backup --rm \
-          -v $(docker volume ls -qf name=^${CMPS_PRJ}_mysql-vol-1$):/backup_mariadb/:rw,z \
-          --entrypoint= \
-          -v ${RESTORE_LOCATION}:/backup:z \
-          ${SQLIMAGE} /bin/bash -c "shopt -s dotglob ; \
-            /bin/rm -rf /backup_mariadb/* ; \
-            /bin/tar -Pxvzf /backup/backup_mariadb.tar.gz"
+          read -p "mailcow will be stopped and the currently active mailcow.conf will be modified to use the DB parameters found in ${RESTORE_LOCATION}/mailcow.conf - do you want to proceed? [Y|n] " MYSQL_STOP_MAILCOW
+          if [[ ${MYSQL_STOP_MAILCOW,,} =~ ^(no|n|N)$ ]]; then
+            echo "OK, skipped."
+            shift
+            continue
+          else
+            echo "Stopping mailcow..."
+            ${COMPOSE_COMMAND} -f ${COMPOSE_FILE} --env-file ${ENV_FILE} down
+          fi
+          #docker stop $(docker ps -qf name=mysql-mailcow)
+          if [[ -d "${RESTORE_LOCATION}/mysql" ]]; then
+          docker run --name mailcow-backup --rm \
+            -v $(docker volume ls -qf name=^${CMPS_PRJ}_mysql-vol-1$):/var/lib/mysql/:rw,z \
+            --entrypoint= \
+            -v ${RESTORE_LOCATION}/mysql:/backup:z \
+            ${SQLIMAGE} /bin/bash -c "shopt -s dotglob ; /bin/rm -rf /var/lib/mysql/* ; rsync -avh --usermap=root:mysql --groupmap=root:mysql /backup/ /var/lib/mysql/"
+          elif [[ -f "${RESTORE_LOCATION}/backup_mysql.gz" ]]; then
+          docker run \
+            -i --name mailcow-backup --rm \
+            -v $(docker volume ls -qf name=^${CMPS_PRJ}_mysql-vol-1$):/var/lib/mysql/:z \
+            --entrypoint= \
+            -u mysql \
+            -v ${RESTORE_LOCATION}:/backup:z \
+            ${SQLIMAGE} /bin/sh -c "mysqld --skip-grant-tables & \
+            until mysqladmin ping; do sleep 3; done && \
+            echo Restoring... && \
+            gunzip < backup/backup_mysql.gz | mysql -uroot && \
+            mysql -uroot -e SHUTDOWN;"
+          elif [[ -f "${RESTORE_LOCATION}/backup_mariadb.tar.gz" ]]; then
+          docker run --name mailcow-backup --rm \
+            -v $(docker volume ls -qf name=^${CMPS_PRJ}_mysql-vol-1$):/backup_mariadb/:rw,z \
+            --entrypoint= \
+            -v ${RESTORE_LOCATION}:/backup:z \
+            ${SQLIMAGE} /bin/bash -c "shopt -s dotglob ; \
+              /bin/rm -rf /backup_mariadb/* ; \
+              /bin/tar -Pxvzf /backup/backup_mariadb.tar.gz"
+          fi
+          echo "Modifying mailcow.conf..."
+          source ${RESTORE_LOCATION}/mailcow.conf
+          sed -i --follow-symlinks "/DBNAME/c\DBNAME=${DBNAME}" ${SCRIPT_DIR}/../mailcow.conf
```

Comment:
sed too (line 30) also check if sed is not busybox, as they have differences in command syntax.

Reply:
Done.